File: LeetCode/remote/excel_sheet_column_title.rb
#   https://leetcode.com/problems/excel-sheet-column-title/
#   Given a positive integer, return its corresponding column title as appear in an Excel sheet.
#   https://leetcode.com/submissions/detail/56369979/
#
#   Submission Details
#   18 / 18 test cases passed.
#       Status: Accepted
#       Runtime: 72 ms
#           
#           Submitted: 0 minutes ago
def convert_to_title(n)
    result = ""
    while n > 0
        n -= 1
        div, mod = n.divmod(26)
        result = ("A".ord + mod).chr + result
        n = div
    end
    
    result
end

File: LeetCode/remote/excel_sheet_column_title.cs
//  https://leetcode.com/problems/excel-sheet-column-title/
//  Fucking hate this problem
//  https://leetcode.com/submissions/detail/60819250/
//
//  Submission Details
//  18 / 18 test cases passed.
//      Status: Accepted
//      Runtime: 48 ms
//          
//          Submitted: 0 minutes ago
public class Solution {
    public string ConvertToTitle(int n) {
        var sb = new StringBuilder();
        while (n != 0)
        {
            n--;
            var div = n / 26;
            var mod = n % 26;
            sb.Append((char)('A' + mod));
            n = div;
        }

        var array = sb.ToString().ToCharArray();
        Array.Reverse(array);
        return new String(array);
    }
}
